lib/rest-adapter.js

```diff
@@ -369,22 +369,40 @@ RestAdapter.errorHandler = function(options) {
         res.statusCode = err.statusCode || err.status || 500;
       }
 
-      debug('Error in %s %s: %s', req.method, req.url, err.stack);
-      var data = {
-        name: err.name,
-        status: res.statusCode,
-        message: err.message || 'An unknown error occurred',
-      };
-
-      for (var prop in err) {
-        data[prop] = err[prop];
+      if (Array.isArray(err)) {
+        var details = err.map(function(it) {
+          var data = generateResponseError(it);
+          delete data.statusCode;
+          return data;
+        });
+
+        err =
+          new Error('Failed with multiple errors, see `details` for more information.');
```

Comment:
Nitpick: I personally prefer this style:

```js
var err = new Error(
  'Failed with multiple errors, see `details` for more information.');
```

or even the following, which works better when the error message needs to be split onto multiple lines:

```js
var msg =  'Failed with multiple errors, see `details` for more information.';
var err = new Error(msg);
```

Feel free to ignore this comment.